server/Tests/BudgetTracker.Business.Tests/UnitTests/BudgetCreateTests.cs

```diff
@@ -0,0 +1,107 @@
+using Bogus;
+using BudgetTracker.Business.Auth;
+using BudgetTracker.Business.Budgeting;
+using BudgetTracker.Business.Ports.Repositories;
+using BudgetTracker.Business.Tests;
+using BudgetTracker.TestUtils.Auth;
+using BudgetTracker.TestUtils.Budgeting;
+using Moq;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace BudgetTracker.Business.Tests.UnitTests
+{
+    public class BudgetCreateTests : BaseUnitTest
+    {
+        private BudgetBuilderFactory<Budget> _budgetBuilderFactory;
+        private UserFactory _userFactory;
+        private Faker _faker;
+
+        public BudgetCreateTests()
+            : base()
+        {
+            _budgetBuilderFactory = GetService<BudgetBuilderFactory<Budget>>();
+            _userFactory = GetService<UserFactory>();
+            _faker = new Faker();
+        }
+
+        [Fact]
+        public async Task Test_BudgetCreated_When_IsRootBudget()
+        {
+            Mock<IBudgetRepository> mockBudgetRepository = new Mock<IBudgetRepository>();
+            User owner = _userFactory.NewUser();
+
+            Budget budget = _budgetBuilderFactory.GetBuilder()
+                                .SetPercentAmount(null)
+                                .SetFixedAmount(47)
+                                .SetOwner(owner)
+                                .Build();
+
+            mockBudgetRepository.Setup(repo => repo.CreateBudget(It.IsAny<Budget>()))
+                                .Returns(Task.FromResult(budget));
+
+            Budget createdBudget = await BudgetCreation.CreateBudgetForUser(budget, owner, mockBudgetRepository.Object);
+
+            mockBudgetRepository.Verify(repo => repo.CreateBudget(budget));
+            Assert.Equal(createdBudget.Name, budget.Name);
+        }
+
+        // (ChildPercentAmount, ChildSetAmount, ParentSetAmount, AmountExpected)
```

Comment:
Haven't seen this used before, love it though. Will this test the test 4 times in a row?

Reply:
Yessir!

Reply:
*dabs*